src/interfaces/noise_model.cpp

```diff
@@ -0,0 +1,19 @@
+#include "gl_depth_sim/interfaces/noise_model.h"
+
+std::vector<float> gl_depth_sim::noise(std::vector<float> distance)
+{
+  float segma;
+  std::random_device r;
+  std::seed_seq seed2{r(), r(), r(), r(), r(), r(), r(), r()};
+  std::mt19937 e2(seed2);
+
+  for (int i = 0 ; i < distance.size() ;++i){
+  // Seed with a real random value, if available
+    segma= 0.0012+0.0019* pow(distance[i]-0.4,2);
```

Comment:
All of these numbers, or `literals`, are of the `double` type. Not a huge deal, but given that you're starting with floats its not unreasonable to stick with the floating point math here. You can change a literal to a float by appending `f`, e.g. `0.0012f`.